### PR TITLE
docs: adding horovod

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding ingo about HorovodStrategy, https://github.com/Lightning-AI/lightning-Horovod

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda

### CODE REVIEW

File: docs/source-pytorch/advanced/third_party/horovod.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+.. code-block:: python
+
+    from lightning import Trainer
+    from lightning_horovod import HorovodStrategy
+
+    # train Horovod on GPU (number of GPUs / machines provided on command-line)
+    trainer = Trainer(strategy="horovod", accelerator="gpu", devices=1)
+
+
```
Comment (Borda) [inline docs/source-pytorch/advanced/third_party/horovod.rst:31]:
```suggestion
This will install both the `Horovod <https://github.com/horovod/horovod#install>`_ package as well as the ``HorovodStrategy`` for the Lightning Trainer.
```
